cmd/performance-profile-creator/cmd/root.go

```diff
@@ -94,7 +96,12 @@ func NewRootCommand() *cobra.Command {
 				return fmt.Errorf("failed to parse the info flag: %v", err)
 			}
 			if info {
-				showClusterData(cluster)
+				// intentionally undocumented and unsupported output to make e2e testing easier.
```

Comment:
I think we should document this as this could be useful if we want to parse the json data for some other purpose. I used it to pipe this into jq for better hierarchical view of the data! I think we should document this as another way to showing info and just mention in the README that if `PERFORMANCE_PROFILE_CREATOR_OUTPUT_JSON=true`  info mode would return an output in json format. By default when  `PERFORMANCE_PROFILE_CREATOR_OUTPUT_JSON` environment variable is not set or when`PERFORMANCE_PROFILE_CREATOR_OUTPUT_JSON=false`  info mode returns a human readable output.

Reply:
if we want to make this official we will need to streamline the UX a lot because suddenly requiring an environment variable to have a different output format is just bad design. We should rather turn `--info` in a value-requiring flag (e.g. `--info=json`, `--info=log`) or maybe even adding a `--format` option.
But there's more: if we support multiple output formats, then it becomes awkward to have an output mode which emits logs (on stderr) vs another which emits JSON (on stdout) which makes the output mode confusing.

Reply:
Yes, I was thinking about that as well. Better to have it in the UX itself! Lets go with --info=json, --info=log option and make log as default. That's what my instinct tells me :) 

Regarding the Json output to STDOUT, on one hand I think that json info to STDOUT should be okay because we either have performance profile or a json info sent to STDOUT on the other hand that I do understand it can be confusing. Maybe we can handle this with proper documentation?

Reply:
I think in info mode totally makes sense to have the output on stdout (that's what stdout is meant to be in the first place :) ).
However
1. it's weird to emit sometimes on stdout (proper output) and sometimes on stderr (because the output... is not error? is the expected output)
2. while this can be documented, I believe this leaves a quite strong "ok, but why?" feeling.

Reply:
Okay let me explain my point of view on this subject:
So if a user runs the following command
```
./build/_output/bin/performance-profile-creator --must-gather-dir-path=testdata/must-gather/must-gather.bare-metal --reserved-cpu-count=20 --rt-kernel=true --mcp-name=master > my-performanceprofile.yaml
```

If it works without error then great we have the performance profile stored in my-performanceprofile.yaml however if we were to send even the errors to STDOUT it would mean that  my-performanceprofile.yaml is storing the error message even without the user knowing about it!! This is something we don't want.

The case of info mode is different as the the user chooses to run the tool in info with json then that provides them the option to send it to a json file to view/use the cluster data . So fine to do that to STDOUT.
 
Hope this helps!